internal/scope/machine.go

```diff
@@ -282,6 +282,39 @@ func (m *MachineScope) GetSSHPublicKey() string {
 	return ""
 }
 
+// GetBasicAuthToken will fetch the BasicAuthSecret on the MvmCluster and
+// and return the token for the given host.
+// If no secret or no value is found, an empty string is returned.
+func (m *MachineScope) GetBasicAuthToken(addr string) string {
+	placement := m.MvmCluster.Spec.Placement
+	if placement.StaticPool == nil || placement.StaticPool.BasicAuthSecret == "" {
+		return ""
+	}
+
+	tokenSecret := &corev1.Secret{}
+	key := types.NamespacedName{
+		Name:      m.MvmCluster.Spec.Placement.StaticPool.BasicAuthSecret,
+		Namespace: m.MvmCluster.Namespace,
+	}
+
+	if err := m.client.Get(m.ctx, key, tokenSecret); err != nil {
+		m.V(2).Info("could not get basicAuthSecret", "error", err) //nolint:gomnd // this magic number is fine
```

Comment:
I'm thinking we should probably bubble up this error to the reconcile function and report it back from there.